app/services/puppet_fact_parser.rb

```diff
@@ -219,7 +219,13 @@ def os_release
       '1.0'
     when /Debian/i
       return "99" if facts[:lsbdistcodename] =~ /sid/
```

Comment:
Should this line be removed/changed? does `lsbdistcodename` exist?

Reply:
I also thought about this. It sometimes exists. It's a legacy fact that's present when `lsb-release` is installed. However, on my machine it returns:
```console
$ facter lsbdistcodename
bullseye
```

I do think it may return sid on a machine that's on unstable instead of testing. This whole function can use a cleanup though.

Reply:
Ok, i guess that's outside the scope of this PR in any case.

Reply:
Yes. Ideally we'd tackle the removal of all legacy facts together. There's https://projects.theforeman.org/issues/27906 for that.